File: app/rcc/scripts/raw_bytes.js
/**
 * Raw Bytes Parser
 *
 * Parses raw binary data into an array of byte values (0-255).
 *
 * INPUT FORMAT: Binary frame data (QByteArray)
 * OUTPUT ARRAY: [0x12, 0x34, 0x56, 0x78, ...]
 *
 * Note: This parser requires Binary (Direct) decoder mode in the Project Editor.
 *       Frame delimiters are automatically removed by Serial Studio.
 */

//------------------------------------------------------------------------------
// Frame Parser Function
//------------------------------------------------------------------------------

/**
 * Parses a raw binary frame into an array of byte values.
 *
 * @param {array} frame - The raw binary data from the data source (byte array)
 * @returns {array} Array of byte values (0-255)
 */
function parse(frame) {
  let dataArray = [];

  for (let i = 0; i < frame.length; ++i) {
    let byte = frame[i];
    dataArray.push(byte);
  }

  return dataArray;
}

File: app/rcc/scripts/lua/raw_bytes.lua
--
-- Raw Bytes Parser
--
-- Returns the byte values from a binary frame table as-is.
-- When the input is a string, returns each byte as its numeric value.
--
-- INPUT (binary): {0x48, 0x65, 0x6C, 0x6C, 0x6F}
-- OUTPUT TABLE:   {"72", "101", "108", "108", "111"}
--
-- INPUT (string): "Hello"
-- OUTPUT TABLE:   {"72", "101", "108", "108", "111"}
--

function parse(frame)
  local result = {}

  if type(frame) == "table" then
    for i = 1, #frame do
      result[i] = frame[i]
    end
  elseif type(frame) == "string" then
    for i = 1, #frame do
      result[i] = frame:byte(i)
    end
  end

  return result
end
